datasetIngestor/sendIngestCommand_test.go

```diff
@@ -0,0 +1,177 @@
+package datasetIngestor
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+	"io"
+)
+
+func TestCreateOrigBlock(t *testing.T) {
+	// Define a slice of Datafile for testing
+	datafiles := []Datafile{
+		{Size: 100},
+		{Size: 200},
+		{Size: 300},
+		{Size: 400},
+	}
+
+	// Call createOrigBlock function
+	block := createOrigBlock(1, 3, datafiles, "test-dataset")
+
+	// Check the Size of the returned FileBlock
+	if block.Size != 500 {
+		t.Errorf("Expected block size of 500, but got %d", block.Size)
+	}
+
+	// Check the length of DataFileList in the returned FileBlock
+	if len(block.DataFileList) != 2 {
+		t.Errorf("Expected 2 datafiles in the block, but got %d", len(block.DataFileList))
+	}
+
+	// Check the DatasetId of the returned FileBlock
+	if block.DatasetId != "test-dataset" {
+		t.Errorf("Expected dataset id of 'test-dataset', but got %s", block.DatasetId)
+	}
+}
+
+func TestSendIngestCommand(t *testing.T) {
+	// Mock HTTP client
+	client := &http.Client{
+		Timeout:   5 * time.Second, // Set a timeout for requests
+		Transport: &http.Transport{
+			// Customize the transport settings if needed (e.g., proxy, TLS config)
+			// For a dummy client, default settings are usually sufficient
+		},
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			// Customize how redirects are handled if needed
+			// For a dummy client, default behavior is usually sufficient
+			return http.ErrUseLastResponse // Use the last response for redirects
+		},
+	}
+
+	// Mock user map
+	user := map[string]string{
```

Comment:
is the user (and the later metadata) needed since you mock the HTTP client?

Reply:
user is a function argument.

Reply:
sorry that was badly written. I wanted to say that user and metadata can have any value here as they are not really used later on, if I understand the test right

Reply:
Yes, their value is not important.

Reply:
should we simplify it then? (happy to keep it as is otherwise)

Reply:
Sure